include/pmacc/eventSystem/events/CudaEvent.cpp

```diff
@@ -68,33 +65,23 @@ namespace pmacc
 
     bool CudaEvent::isFinished()
     {
-        // avoid cupla driver calls if event is already finished
-        if(finished)
-            return true;
-        assert(isRecorded);
-
-        cuplaError_t rc = cuplaEventQuery(event);
-
-        if(rc == cuplaSuccess)
+        // avoid alpaka calls if event is already finished
+        if(!finished)
         {
-            finished = true;
-            return true;
+            assert(stream);
```

Comment:
I would personally prefer the more explicit version
```suggestion
            assert(stream.has_value());
```
but it's not a strong opinion.